src/StreamJsonRpc/ActivityTracingStrategy.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc
+{
+    using System;
+    using System.Diagnostics;
+    using Microsoft;
+    using StreamJsonRpc.Protocol;
+
+    /// <summary>
+    /// Synchronizes activities as set by the <see cref="Activity"/> class over RPC.
+    /// </summary>
+    /// <seealso cref="CorrelationManagerTracingStrategy"/>
+    public class ActivityTracingStrategy : IActivityTracingStrategy
+    {
+        /// <inheritdoc/>
+        public void ApplyOutboundActivity(JsonRpcRequest request)
+        {
+            Requires.NotNull(request, nameof(request));
+
+            if (Activity.Current?.IdFormat == ActivityIdFormat.W3C)
+            {
+                request.TraceParent = Activity.Current.Id;
+                request.TraceState = Activity.Current.TraceStateString;
+            }
+        }
+
+        /// <inheritdoc/>
+        public IDisposable? ApplyInboundActivity(JsonRpcRequest request)
+        {
+            Requires.NotNull(request, nameof(request));
+
+            if (request.TraceParent is object)
+            {
+                var state = new State(new Activity(request.Method));
+                state.NewActivity.TraceStateString = request.TraceState;
+                state.NewActivity.SetParentId(request.TraceParent);
+                state.NewActivity.Start();
+                return state;
+            }
+
+            return null;
```

Comment:
In the case inbound activity's trace parent is not an object, should we consider to start a new activity and set the parent to null/ emty? so for all JsonRpcRequest calls from here will have the parentId set to this new activity?

Reply:
Good idea. That improves the diagnosability on the server-side when the client hasn't gone to any trouble to set a traceparent. I've made the change.